modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/component/ImprovedSearchPodcastResultRow.kt

```diff
@@ -0,0 +1,146 @@
+import au.com.shiftyjelly.pocketcasts.models.to.FolderItem
+import au.com.shiftyjelly.pocketcasts.models.to.SearchAutoCompleteItem
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+
+@Composable
+fun ImprovedSearchPodcastResultRow(
+    item: SearchAutoCompleteItem.Podcast,
+    onClick: () -> Unit,
+    onFollow: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    ImprovedSearchPodcastResultRow(
+        podcastUuid = item.uuid,
+        title = item.title,
+        author = item.author,
+        isSubscribed = item.isSubscribed,
+        onClick = onClick,
+        onFollow = onFollow,
+        modifier = modifier,
+    )
+}
+
+@Composable
+fun ImprovedSearchPodcastResultRow(
+    folderItem: FolderItem,
+    onClick: () -> Unit,
+    onFollow: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    when (folderItem) {
+        is FolderItem.Podcast ->
+            ImprovedSearchPodcastResultRow(
+                podcastUuid = folderItem.podcast.uuid,
+                title = folderItem.podcast.title,
+                author = folderItem.podcast.author,
+                isSubscribed = folderItem.podcast.isSubscribed,
+                onClick = onClick,
+                onFollow = onFollow,
+                modifier = modifier,
+            )
+
+        is FolderItem.Folder ->
+            SearchFolderRow(
+                folder = folderItem.folder,
+                podcasts = folderItem.podcasts,
+                onClick = onClick,
+                modifier = modifier,
+            )
+    }
+}
+
+@Composable
+private fun ImprovedSearchPodcastResultRow(
+    podcastUuid: String,
+    title: String,
+    author: String,
+    isSubscribed: Boolean,
+    onClick: () -> Unit,
+    onFollow: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier
+            .clickable(onClick = onClick)
+            .padding(horizontal = 16.dp, vertical = 8.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(12.dp),
+    ) {
+        PodcastImage(
+            uuid = podcastUuid,
+            elevation = 6.dp,
+            cornerSize = 4.dp,
+        )
+        Column(modifier = Modifier.weight(1f)) {
+            TextH40(
+                text = title,
+                color = MaterialTheme.theme.colors.primaryText01,
+                maxLines = 1,
+            )
+            TextP50(
+                text = author,
+                color = MaterialTheme.theme.colors.primaryText02,
+                maxLines = 1,
+            )
+        }
+        if (isSubscribed) {
+            Icon(
+                modifier = Modifier
+                    .minimumInteractiveComponentSize(),
+                painter = painterResource(IR.drawable.ic_check),
+                contentDescription = null,
+                tint = MaterialTheme.theme.colors.support02,
+            )
```

Comment:
Should we make this tick a bit bigger, similar to the Discover section?

<img width="2436" height="2531" alt="Image" src="https://github.com/user-attachments/assets/d6e7a8b1-0c20-406c-a229-d30914d72063" />